### DISCUSSION TITLE
Index + Thumbparams not working?

### DISCUSSION THREAD

**Discussion by NotAkitake** [I5UCC/VRC-ASL_Gestures#5] [General]
Heya, I recently tried your installer on a couple of new models, for Index with Thumbparams, and it always ends up being the same hand shape, the claw, but I can't get anything else to work.

And yes I do have thumbparams running with osc on, and it works for other avatars (not mine, and not your package) with thumbparams too.

**Comment by I5UCC** [comment]
Hey!
I took some time to try to reproduce the problem you are having, tho even on a new model, it still seems to be fine always

Ok so, the "claw hand" animation usually means that an animation is missing, or something is overriding it. 
so make sure that all of the animations in the layer do still exist and nothing is overriding it (that could be the case when the installer is overriding an already exisiting Gesture layer) and the masks for each layer are set.

Otherwise, when testing in game, using the debug window(not the osc one) can you see the layer changing, depending on where you position your thumb? and also ofc the parameters LeftThumb and RightThumb changing values?

Also might be dumb but doesnt sound like this is the problem, when using my installer, the parameter "SignsDisabled" is set to True by default. so make sure in game you dont have it disabled, as it is by default. (i might wanna change that behaviour)

If other packages work, and they require you to do it manually, try going the manual route with my package too and maybe that helps.
Also, it shouldnt make problems, but make sure that Write Defaults are either on or off on everything, sometimes weird and unexpected behaviour occours if its mixed. My package is completely Write Defaults disabled.

**Reply by I5UCC** [reply]
Generally, my animators are not very different from others that ive seen so far in how they are built, besides maybe having Write Defaults off by default and also having a Toggle for it.

**Reply by I5UCC** [reply]
Also if you had no issues with the past Animators that i did that were using Anystate transitions, they still exist under the "OLD_AnyState_Animators" folder. Those have to be installled manually.

If you still have issues, maybe add me on Discord so i can help you further, I5UCC#6781 or just continue here.

**Reply by NotAkitake** [reply]
The model I'm testing on doesn't even have a gesture layer by default, so your installer went straight to using the ASLGestures_Index_Thumbparams one.

Which has the Base Layer, Left Hand and Right Hand, although since your recent updates I'm not even sure how your gesture layer works because I don't see mentions of animations anywhere.

![image](https://user-images.githubusercontent.com/7873528/191335138-c137f61d-7ca9-4570-80f0-2ba690631d57.png)
![image](https://user-images.githubusercontent.com/7873528/191335237-a0beacd3-5f6f-42d6-ad4c-b0d6985fa6f2.png)
![image](https://user-images.githubusercontent.com/7873528/191335275-2c0e08a8-58c2-447a-a6eb-623e90f60b4d.png)
![image](https://user-images.githubusercontent.com/7873528/191335316-6e4dcab1-9d9d-41e0-a4cc-dcd9d819a59d.png)

I'll look into doing things manually or differently, not sure what's wrong here as I'm less familiar with your newer versions.

**Reply by I5UCC** [reply]
The states saying "Neutral", "Fist" etc. are so called Sub-StateMachines. Double clicking on any of them will give you something more familiar.
Otherwise what i see is that your avatar descriptor has two slots for FX(where should be one "Action" layer and one "FX" layer only, that shouldnt be the case and is generally an error done while importing. Resetting it to default then setting the layers again could already help.
